cluster/clientbin.sh

```diff
@@ -68,7 +68,8 @@ esac
 # Get the absolute path of the directory component of a file, i.e. the
 # absolute path of the dirname of $1.
 get_absolute_dirname() {
-  echo "$(cd "$(dirname "$1")" && pwd)"
+  absolute_dirname=$(cd "$(dirname "$1")" && pwd)
```

Comment:
Not clear what problem this is trying to solve

Also would a bash subshell help here?

Reply:
The shellcheck failure is:
```
In cluster/clientbin.sh line 71:
  echo "$(cd "$(dirname "$1")" && pwd)"
       ^-- SC2005: Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.
```

But if we just remove `echo`, use `cd "$(dirname "$1")" && pwd`, the working directory will change.
The behavior of the script changes. And that's not what we want.
`$(cd "$(dirname "$1")" && pwd)` will not change the working directory.

Reply:
thanks, this is a reasonable enough solution.

Reply:
The CD is local to the subshell
```console
$  p=$(pwd) && ( cd src ) && q=$(pwd) && [[ $p == $q ]] && echo yes || echo no
yes
```

Reply:
Yes! 
```
$ p=$(pwd) && cd /tmp && q=$(pwd) && [[ $p == $q ]] && echo yes || echo no
no

$ p=$(pwd) && ( cd /tmp ) && q=$(pwd) && [[ $p == $q ]] && echo yes || echo no
yes

$ p=$(pwd) && $( cd /tmp ) && q=$(pwd) && [[ $p == $q ]] && echo yes || echo no
yes
```

Reply:
```bash
get_abs_name() {
  ( cd /tmp && pwd )
}
```

I believe will accomplish the same thing, no need for the var and echo

Reply:
That makes the code cleaner. Thank you!